scripts/main-loop.js

```diff
@@ -176,20 +176,27 @@ define(['./packet', './loop-factory', './data-factory', './device-prop-codes'],
                 hexString = "48 00 00 00 00 00 00 00" + // 72 distinct properties seems too high?
 
                     "05 50" + // white balance
-                    "04 00 01 01 00 00" +
-                    "02 00 02 0f 00 02 00 04 00 11 80 10 80 06 00 01" +
-                    "80 02 80 03 80 04 80 07 00 30 80 12 80 20 80 21" +
-                    "80 22 80 0f 00 02 00 04 00 11 80 10 80 06 00 01" +
-                    "80 02 80 03 80 04 80 07 00 30 80 12 80 20 80 21" +
-                    "80 22 80" +
+                    "04 00" + // data type. uint16 (4 bytes)
+                    "01 01 00 00" + // get and set both available
+                    "02 00" + // current val
+                    "02" + // enumeration
+                    "0f 00" + // 15 values in array
+                    "02 00 04 00 11 80 10 80 06 00 01 80 02 80" +
+                    "03 80 04 80 07 00 30 80 12 80 20 80 21 80" +
+                    "22 80"
+                    "0f 00" + // 15 values in array
+                    "02 00 04 00 11 80 10 80 06 00 01 80 02 80" +
+                    "03 80 04 80 07 00 30 80 12 80 20 80 21 80" +
+                    "22 80" +
 
                     "07 50" + // f number
                     "04 00" + // data type. uint16
-                    "01 01 ff ff" + // get/set (uint8), unknown (uint8), factory value (data type)
+                    // get (01 = available, 02 = unavailable), set (01 = available, 02 = unavailable)
+                    "01 01 ff ff" + // unknown (uint8), factory value (data type)
                     dataFactory.createWord(serverState.fNumber).toHex() + // current val
                     "02" + // enumeration (1=range, 2=enum)
-                    "13 00" + // 19 values in array @simon: 1 array will be "available" and one will be "supported".
-                    "18 01 40 01 5e 01 90 01 c2 01 f4 01 30 02 76 02" + // list starts with 280 (F2.8)
+                    "13 00" + // 19 values in array @simon: 1 array will be "available" and one will be "supported" not able to confirm this.
```

Comment:
You're probably right given it's how it used to work in the old HTTP API. Would be great to find an example where these two lists are not identical.

 I'm a bit surprised that shutter speed has 55 elements in both lists, even though we are in aperture-priority mode (`set` flag is set to `02` though).

Reply:
Flash mode below has 5 in the second array and 3 in the first :) The ones missing from the first array are "Auto" and "Off"... I'm thinking that this means that the second array is supported, and the first is available. Will be interesting to see if there are any further examples throughout the data! It's a bit tricky to tell because if you disable values in either of them they disappear from Sony's UI. However if the first array is longer than the second Sony's parsing seems to break and the full configured UI isn't shown which makes me think my above assumption is correct!